spin/pty.go

```diff
@@ -0,0 +1,30 @@
+package spin
+
+import (
+	"os"
+
+	"github.com/charmbracelet/x/term"
+	ptytty "github.com/creack/pty"
```

Comment:
Fyi, this wouldn't work on Windows :/ We should use our `x/xpty` instead for a cross-platform solution

Reply:
yea, i just saw it going to say the same, is `x/pty` conpty?

Reply:
We have `x/xpty` which uses `creack/pty` and our `x/conpty` to provide a cross-platform API

Reply:
got it! thanks @aymanbagabas